sqlglot/schema.py

```diff
@@ -0,0 +1,297 @@
+import abc
+
+from sqlglot import exp
+from sqlglot.errors import OptimizeError
+from sqlglot.helper import csv_reader
+
+
+class Schema(abc.ABC):
+    """Abstract base class for database schemas"""
+
+    @abc.abstractmethod
+    def column_names(self, table, only_visible=False):
+        """
+        Get the column names for a table.
+        Args:
+            table (sqlglot.expressions.Table): Table expression instance
+            only_visible (bool): Whether to include invisible columns
+        Returns:
+            list[str]: list of column names
+        """
+
+    @abc.abstractmethod
+    def get_column_type(self, table, column):
+        """
+        Get the exp.DataType type of a column in the schema.
+
+        Args:
+            table (sqlglot.expressions.Table): The source table.
+            column (sqlglot.expressions.Column): The target column.
+        Returns:
+            sqlglot.expressions.DataType.Type: The resulting column type.
+        """
+
+
+class MutableSchema(Schema):
+    """
+    Based on MappingSchema but mutable so you can add tables over time.
```

Comment:
Strange that the subclass is "based on the MappingSchema", a superclass.

I think this class should be a lot simpler. It should be more of an interface.

Reply:
i would have expected the default Mapping Schema to come first

Reply:
I think that docstring was just not updated after the refactor. 

Mapping schema did come first before the last request for changes. The last request asked for this hierarchy of Schema -> MutableSchema(Schema) -> MappingSchema(MutableSchema)

Reply:
Right now, all `MappingSchema` does is _unimplement_ add_table. That's silly.

I think it's fine for the the MappingSchema to implement add_table. I just didn't think add_table should be part of the main Schema interface.

We only need this intermediate interface, MutableSchema, if the DataFrame API wants to support custom Schema types.